docs/known_issues.rst

```diff
@@ -347,3 +347,24 @@ processing and you just want docstrings to work, this may be
 acceptable.
 
 The IPython issue: https://github.com/ipython/ipython/pull/2738
+
+Compatibility issues with pytest 3.7 and later
+----------------------------------------------
+
+Due to a bug in `pytest <http://www.pytest.org>`_ related to test collection,
+the tests for the core astropy package for version 2.0.x (LTS), and for packages
+using the core package's test infrastructure and being tested against 2.0.x
+(LTS) will not be executed correctly with pytest 3.7, 3.8, or 3.9, and therefore
+pytest 3.6 or earlier, or pytest 4.0 or later should be used. The symptom of
+this bug is that no tests or only tests in rst files are collected. Note that
+this is not a problem in version 3.0.x and above for the core package.
+
+However, there is also an unrelated issue with pytest 4.0 and later which can
+also cause issues when collecting tests - in this case, the symptom is that the
+test collection hangs and/or appears to run the tests recursively. If you are
+maintaining a package that was created using the astropy
+`package template <http://github.com/astropy/package-template>`_, then
+this can be fixed by updating to the latest version of the ``_astropy_init.py``
+file. The root cause of this issue is that pytest now tries to pick up the
+top-level ``test()`` function as a test, so we need to make sure that we set a
+``test.__test__`` attribute on the function to ``False``.
```

Comment:
Could you add the third issue as well namely than in python 4.0+ the deprecation warning for `pytest_plugins` has turned into an error, so the LTS branch can never be pytest 4.0 compatible. Branch 3.0.x might be also affected, I'm not sure.

I think adding `pytest-astropy` and thus the astropy plugins as separate packages would fix this in addition of switching to implicit imports from `astropy.tests.pytest_plugins` (the latter may not be needed, I'm not sure any more given that I was messing with a combination of all of the issues above)


https://github.com/astropy/astropy/issues/7782